scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -282,6 +285,42 @@ static void dispatch_local(struct task_struct *p, u64 enq_flags)
 	scx_bpf_dispatch(p, SCX_DSQ_LOCAL, slice_ns, enq_flags | SCX_ENQ_LOCAL);
 }
 
+/*
+ * Get a valid CPU to dispatch a task, starting with @cpu and checking across
+ * all the available CPUs in the system.
+ */
+static s32 _get_task_cpu_slow(struct task_struct *p, s32 cpu)
+{
+	int i;
+
+	for (i = 0; i < num_possible_cpus; i++) {
+		cpu = (cpu + i) % num_possible_cpus;
+		if (bpf_cpumask_test_cpu(cpu, p->cpus_ptr))
+			return cpu;
+	}
+
+	return -1;
```

Comment:
Something else you could do is call `bpf_cpumask_any_distribute(p->cpus_ptr)` to choose a random CPU from `p->cpus_ptr`? Note that it returns a cpu >= num_cpus if none of the bits are set, so you'd have to do something like:
```
cpu = bpf_cpumask_any_distribute(p->cpus_ptr);
return cpu < num_possible_cpus ? cpu : -1;
```

Reply:
Nice suggestion! Thanks! I'll apply it and run some tests. I think we should still maintain the logic to try the prev_cpu before picking a random one using `bpf_cpumask_any_distribute()`, that should reduce some unnecessary migrations, but I'll test/measure both and check if one shows some clear benefits vs the other.

Reply:
Yep, agreed. I was thinking that `bpf_cpumask_any_distribute()` could replace that for loop. But trying to choose prev_cpu first seems like a good idea.